sw/device/lib/dif/dif_gpio.h

```diff
@@ -0,0 +1,327 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_LIB_DIF_GPIO_H
+#define SW_DEVICE_LIB_DIF_GPIO_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+
+/**
+ * Configuration for initializing a GPIO device.
+ */
+typedef struct dif_gpio_config { mmio_region_t base_addr; } dif_gpio_config_t;
+
+/**
+ * Internal state of a GPIO device.
+ *
+ * Instances of this struct must be initialized by |dif_gpio_init()| before
+ * being passed to other functions.
+ */
+typedef struct dif_gpio { mmio_region_t base_addr; } dif_gpio_t;
+
+/**
+ * Generic return codes for the functions in this library.
+ */
+typedef enum dif_gpio_result {
+  kDifGpioResultOK,
+  kDifGpioResultInvalidArgument,
+} dif_gpio_result_t;
+
+/**
+ * Set of allowed interrupt trigger configurations.
+ */
+typedef enum dif_gpio_irq_trigger {
+  kDifGpioIrqEdgeRising,
+  kDifGpioIrqEdgeFalling,
+  kDifGpioIrqLevelLow,
+  kDifGpioIrqLevelHigh,
+  kDifGpioIrqEdgeRisingFalling,
+  kDifGpioIrqEdgeRisingLevelLow,
+  kDifGpioIrqEdgeFallingLevelHigh,
+} dif_gpio_irq_trigger_t;
```

Comment:
Elements should be named after their type, so `kDifGpioIrqTriggerEdgeRising`, for example.

Reply:
Thanks! Dropped `_trigger` from the type name to keep the variants short.

Reply:
Should we add this to the DIF styling guidelines or OT style guidelines? As mentioned [here](https://github.com/lowRISC/opentitan/pull/1320/files/007fc1c3be8ca24b1b5434a6b32dfa19e575592a#r374214254) I do not believe this is covered, and isn't the naming scheme used in the example given in the standard.

Reply:
I think this might be a general C style guide thing. It may have been skipped by the google c++ style guide as c++ has namespaces which can cover this issue.